back/engines/commercial/machine_translations/app/controllers/machine_translations/web_api/v1/machine_translations_controller.rb

```diff
@@ -32,6 +32,12 @@ def show
             begin
               @translation = MachineTranslationService.new.build_translation_for @translation_attributes
               authorize @translation
+            
+            # Occasional SQL-injection attacks have been attempted via values for attribute_name,
+            # in which case we ensure @translation == nil, and rescue the resulting Pundit error.
+            rescue Pundit::NotDefinedError
```

Comment:
Pundit doesn't like authorising `nil`, so let's handle the predictable error in this case to return status 422 (avoiding the Sentry error).

Reply:
> Currently, it's not super clear to me what is the reason for `Pundit::NotDefinedError` without the comment. But the comments in general are not DRY and getting outdated overtime, so maybe we can try to be more explicit with the code.
> 
> ```ruby
> @translation = MachineTranslationService.new.build_translation_for @translation_attributes
> if @translation.nil?
>   render json: { errors: { base: [{ error: 'unable_to_translate' }] } }, status: :unprocessable_entity
>   return
> end
> authorize @translation
> ```

Pretty sure I tried this and a different Pundit Error was raised. I'll try again, though.

Reply:
Yes, indeed, your suggested approach (which i had tried also), raises
```
Pundit::AuthorizationNotPerformedError
```
Though I don't understand how Pundit is involved, given we return if `@translation.nil?`

Reply:
You can read more about how to handle it here https://github.com/varvet/pundit#conditional-verification

Reply:
Thanks, but I simply don't understand why there is a Pundit exception to handle after we should have already returned from the method `if @translation.nil?` (using the code you suggest here).

Reply:
OK, adding `skip_authorization` works... so I guess authorization actually happens after the method returns. :thinking:

Reply:
> authorization actually happens after the method returns

Not really. We just configured it this way so it raises an exception if we didn't authorize in the action
https://github.com/CitizenLabDotCo/citizenlab/blob/99b72bcc8128c0ef5e66ceab7065c15366724dee/back/app/controllers/application_controller.rb#L8

You can read about it here https://github.com/varvet/pundit#ensuring-policies-and-scopes-are-used

Reply:
Ahh, I see. Thanks!